### PR TITLE
Dapp Browser + provider

### PR DESCRIPTION
**Pull request by brunobar79**
Fixes APP-####

## Screen recordings / screenshots
https://recordit.co/TRKYEIOouD

### CODE REVIEW

File: src/components/DappBrowser/handleProviderRequest.ts [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+
+const getChain = (chainId: number) => RainbowNetworks.find(network => Number(network.id) === chainId);
+
+const getProvider = ({ chainId }: { chainId?: number | undefined }) => {
+  const network = getNetworkFromChainId(chainId || 1);
+  return getCachedProviderForNetwork(network) as unknown as Provider;
+};
+
+const checkRateLimitFn = async (host: string) => {
+  // try {
```
Comment (brunobar79) [inline src/components/DappBrowser/handleProviderRequest.ts:175]: I'll hook this up later

File: src/components/DappBrowser/handleProviderRequest.ts [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+  }: {
+    proposedChain: AddEthereumChainProposedChain;
+    callbackOptions?: CallbackOptions;
+  }) => {
+    const { chainId } = proposedChain;
+    const supportedChains = RainbowNetworks.filter(network => network.features.walletconnect).map(network => network.id.toString());
+    const numericChainId = convertHexToString(chainId);
+    const supportedChainId = supportedChains.includes(numericChainId);
+    alert('Chain Id not supported');
+    console.warn('PROVIDER TODO: TODO SEND NOTIFICATION');
```
Comment (brunobar79) [inline src/components/DappBrowser/handleProviderRequest.ts:291]: left this so we can add some toast later on

File: src/components/DappBrowser/handleProviderRequest.ts [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+    const numericChainId = convertHexToString(chainId);
+    const supportedChainId = supportedChains.includes(numericChainId);
+    if (supportedChainId) {
+      const host = getDappHost(callbackOptions?.sender.url) || '';
+      const activeSession = getActiveSession({ host });
+      if (activeSession) {
+        appSessionsStore.getState().updateActiveSessionNetwork({ host: host, network: getNetworkFromChainId(Number(numericChainId)) });
+        messenger.send(`chainChanged:${host}`, Number(numericChainId));
+      }
+      console.warn('PROVIDER TODO: TODO SEND NOTIFICATION');
```
Comment (brunobar79) [inline src/components/DappBrowser/handleProviderRequest.ts:322]: left this so we can add some toast later on

File: src/components/DappBrowser/handleProviderRequest.ts [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+    });
+  }
+
+  if (!response) {
+    throw new UserRejectedRequestError(Error('User rejected the request.'));
+  }
+  return response;
+};
+
+const isSupportedChainId = (chainId: number) => RainbowNetworks.filter(network => Number(network.id) === chainId).length > 0;
```
Comment (skylarbarrera) [inline src/components/DappBrowser/handleProviderRequest.ts:154]: i feel like this should be

const isSupportedChainId = (chainId: number) => !!RainbowNetworks.find(network => Number(network.id) === chainId);

File: src/utils/requestNavigationHandlers.ts [hunk 5]
```diff
@@ -24,9 +24,9 @@ export type RequestType = 'walletconnect' | 'browser';
 // Dapp Browser
 
 export interface DappConnectionData {
-  dappName: string;
+  dappName?: string;
```
Comment (skylarbarrera) [inline src/utils/requestNavigationHandlers.ts:27]: curious when this would be undefined

File: src/utils/requestNavigationHandlers.ts [hunk 6]
```diff
@@ -71,6 +71,7 @@ export const handleDappBrowserConnectionPrompt = (dappData: DappConnectionData):
 export const handleDappBrowserRequest = async (request: Omit<RequestData, 'displayDetails'>): Promise<string | Error> => {
   const nativeCurrency = store.getState().settings.nativeCurrency;
   const displayDetails = getRequestDisplayDetails(request.payload, nativeCurrency, request.network);
+  console.log('displayDetails', displayDetails);
```
Comment (skylarbarrera) [inline src/utils/requestNavigationHandlers.ts]: insert spiderman meme

**Review by skylarbarrera — APPROVED** [review]
lets get it lets go

File: src/utils/requestNavigationHandlers.ts (hunk 5 above)
Comment (skylarbarrera) [reply src/utils/requestNavigationHandlers.ts:27]: prob ts tbh

Comment (brunobar79) [reply src/utils/requestNavigationHandlers.ts:27]: a website without <title> tag

File: src/components/DappBrowser/handleProviderRequest.ts (hunk 4 above)
Comment (brunobar79) [reply src/components/DappBrowser/handleProviderRequest.ts:154]: I feel like there shouldn't be networks at all but ok

Comment (skylarbarrera) [reply src/components/DappBrowser/handleProviderRequest.ts]: insert spiderman meme

**Pull request merged**